### PR TITLE
CORE-9100 Proxy Logrus stdout/stderr logging

### PR DESCRIPTION
**Pull request by psarando**
This PR will proxy `logrus` stdout/stderr logging for docker compose commands, and will also update `logrus` to the latest version.

Proxying the Logrus Writer through `fmt.Fprintf` can prevent "token too long" errors if a docker command generates a lot of output (from pulling many input containers at once, for example) and Logrus attempts to log all of that output in one log line. The `fmt.Fprintf` proxy will break up the output into multiple log lines.

For example, when a job runs with 1000's of input files, this bug produced a `script-error.log` like the following:
```
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"Pulling input_1 ... ","service":"road-runner","time":"2018-02-07T14:36:50-07:00"}
...
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"Pulling input_4096 ... ","service":"road-runner","time":"2018-02-07T14:36:50-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"error","msg":"Error while reading from Writer: bufio.Scanner: token too long","service":"road-runner","time":"2018-02-07T14:41:20-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"error","msg":"signal: broken pipe","service":"road-runner","time":"2018-02-07T14:41:21-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"Beginning to upload outputs to ...
```

This update will produce a `script-error.log` like the following:
```
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"Pulling input_1 ... ","service":"road-runner","time":"2018-02-09T10:49:35-07:00"}
...
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"Pulling input_4096 .","service":"road-runner","time":"2018-02-09T10:50:13-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"\u001b[3331A\u001b[2K\rPulling input_3722 ... \u001b[32mdone\u001b[0m\r\u001b[3331B\u001b[3990A\u001b[2K\rPulling input_2876 ... \u001b[32mdone\u001b[0m\r\u001b[3990B\u001b[2921A\u001b[2K\rPulling input_1024 ... \u001b[32mdone\u001b[0m\r\u001b[2921B\u001b[3330A\u001b[2K\rPulling input_3725 ... \u001b[32mdone\u001b[0m\r\u001b[3330B\u001b[3814A\u001b[2K\rPulling input_347 ... \u001b[32mdone\u001b[0m\r\u001b[3814B","service":"road-runner","time":"2018-02-09T10:50:13-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"\u001b[1310A\u001b[2K\rPulling input_1806 ... \u001b[32mdone\u001b[0m\r\u001b[1310B","service":"road-runner","time":"2018-02-09T10:50:13-07:00"}
{"art-id":"road-runner","group":"org.cyverse","level":"info","msg":"\u001b[3246A\u001b[2K\rPulling input_2582 ... \u001b[32mdone\u001b[0m\r\u001b[3246B","service":"road-runner","time":"2018-02-09T10:50:13-07:00"}
...
```
So this update prevents `logrus` from logging all the `docker pull` progress output, for every input container, in a single log entry.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
I'm not quite sure I understand what the before and after look like here (other than "error" and "no longer error" of course) -- are we splitting up the message between several lines, with the JSON wrapping around each of them, or are we wrapping the message in the JSON stuff and then adding linebreaks in the middle of that, or some other sort of adding-linebreaks that I'm missing? We aren't currently ingesting these messages anywhere, so it's not like we're going to break anything like that, but I'm still wondering what to expect.

**Comment by psarando** [comment]
Sorry about that. I've updated this PR's description with examples of the logging output before and after the fix.

File: run.go [hunk 1]
```diff
@@ -14,11 +14,29 @@ import (
...
+// "Error while reading from Writer: bufio.Scanner: token too long" errors
+// if a docker command generates a lot of output
+// (from pulling many input containers at once, for example)
+// and Logrus attempts to log all of that output in one log line.
+type logrusProxyWriter struct {
+	entry *logrus.Entry
+}
+
+func (w *logrusProxyWriter) Write(b []byte) (int, error) {
+	return fmt.Fprintf(w.entry.Writer(), string(b))
```
Comment (psarando) [inline run.go:33]: I'm not exactly sure why `logrus` was logging all the `docker pull` output into 1 log entry before, but proxying through `fmt.Fprintf` (and also converting the bytes to a `string` first) seems to break up the logged output into 1 line for each newline in the `docker pull` output.

**Comment by ianmcorvidae** [comment]
Huh, interesting. I guess it's probably not sure what to do with the bare carriage returns and such.

In any case, I think this looks pretty reasonable. I partly wonder if it'll still run into issues if the stars align right, but I think this is good to ship in any case.

**Comment by psarando** [comment]
Thanks for the review 👍

**Pull request merged**